client/components/pages/EditProfilePage.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import toastr from 'toastr';
+import SignupForm from '../forms/SignupForm.jsx';
+import handleError from '../../utils/errorHandler';
+import { signupValidator } from '../../utils/validator';
+import { updateUser } from '../../actions/userActions';
+import Sidebar from '../layouts/Sidebar.jsx';
+
+class EditProfilePage extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleConfirmPassword = this.handleConfirmPassword.bind(this);
+    this.state = {
+      confirmPassword: '',
+      signupErrors: {},
+      signupDetails: {
+        name: '',
+        email: '',
+        username: '',
+        password: ''
+      }
+    };
+  }
+
+  handleChange(event) {
+    const signupDetails = this.state.signupDetails;
+    signupDetails[event.target.name] = event.target.value.substr(0, 30);
+    this.setState({ signupDetails });
```

Comment:
I guess you are trying to put a 30character limit. A regular text `<input />` can do that. It can be a child component of SignupForm